ArduCopter/tuning.cpp

```diff
@@ -7,23 +7,23 @@
 
 // tuning - updates parameters based on the ch6 tuning knob's position
 //  should be called at 3.3hz
-void Copter::tuning() {
-    RC_Channel *rc6 = rc().channel(CH_6);
+void Copter::tuning()
+{
+    const RC_Channel *rc6 = rc().channel(CH_6);
 
-    // exit immediately if not using tuning function, or when radio failsafe is invoked, so tuning values are not set to zero
-    if ((g.radio_tuning <= 0) || failsafe.radio || failsafe.radio_counter != 0 || rc6->get_radio_in() == 0) {
+    // exit immediately not using tuning function or min and max are both zero
+    if ((g.radio_tuning <= 0) || (is_zero(g2.tuning_min.get()) && is_zero(g2.tuning_max.get()))) {
         return;
     }
 
-    uint16_t radio_in = rc6->get_radio_in();
-    float v = constrain_float((radio_in - rc6->get_radio_min()) / float(rc6->get_radio_max() - rc6->get_radio_min()), 0, 1);
-    int16_t control_in = g.radio_tuning_low + v * (g.radio_tuning_high - g.radio_tuning_low);
-    float tuning_value = control_in / 1000.0f;
-    
-    // Tuning Value should never be outside the bounds of the specified low and high value
-    tuning_value = constrain_float(tuning_value, g.radio_tuning_low/1000.0f, g.radio_tuning_high/1000.0f);
+    // exit immediately when radio failsafe is invoked or transmitter has not been turned on
+    if (failsafe.radio || failsafe.radio_counter != 0 || rc6->get_radio_in() == 0) {
+        return;
+    }
 
-    Log_Write_Parameter_Tuning(g.radio_tuning, tuning_value, control_in, g.radio_tuning_low, g.radio_tuning_high);
+    const uint16_t radio_in = rc6->get_radio_in();
+    float tuning_value = linear_interpolate(g2.tuning_min, g2.tuning_max, radio_in, rc6->get_radio_min(), rc6->get_radio_max());
```

Comment:
`const`

Reply:
we can't actually make it const because we modify it in the switch statement case for the winch.. I could instead take a copy in that case statement to allow this to be const..